### ISSUE TITLE
[graphql] Error running query: argument of type 'NoneType' is not iterable

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-flow#3855]
**Describe the bug**

Trying a new query to retrieve the offline data, used in the Cylc UI (for the offline UI, https://github.com/cylc/cylc-ui/pull/505).

```
{
  "errors": [
    {
      "message": "argument of type 'NoneType' is not iterable",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "workflows"
      ]
    }
  ],
  "data": {
    "workflows": null
  }
}
```

**Release version(s) and/or repository branch(es) affected?**

`master` for UIS (01988bb66307e53a1bda66e9ead70c64e0cda9c9) and Flow (bd293959edcf6db96414d608ef43772299514e36)

**Steps to reproduce the bug**

Run the query in GraphiQL, with a workflow on hold.

**Expected behavior**

Valid data retrieved, or an error response with the explanation.

**Screenshots**

**Additional context**

Query

```
query ($workflowID: ID) {
  workflows (ids: [$workflowID]) {
    ...WorkflowData
    cyclePoints: familyProxies (ids: ["root"], ghosts: true) {
      ...CyclePointData
    }
    taskProxies (sort: { keys: ["name"], reverse: false }, ghosts: true) {
      ...TaskProxyData
      jobs(sort: { keys: ["submit_num"], reverse:true }) {
        ...JobData
      }
    }
    familyProxies (exids: ["root"], sort: { keys: ["name"] }, ghosts: true) {
      ...FamilyProxyData
    }
  }
}

fragment WorkflowData on Workflow {
  id
  name
  status
  owner
  host
  port
}

fragment CyclePointData on FamilyProxy {
  id
  cyclePoint
}

fragment FamilyProxyData on FamilyProxy {
  id
  name
  state
  cyclePoint
  firstParent {
    id
    name
    cyclePoint
    state
  }
}

fragment TaskProxyData on TaskProxy {
  id
  name
  state
  isHeld
  cyclePoint
  latestMessage
  firstParent {
    id
    name
    cyclePoint
    state
  }
  task {
    meanElapsedTime
    name
  }
}

fragment JobData on Job {
  id
  firstParent: taskProxy {
    id
  }
  batchSysName
  batchSysJobId
  host
  startedTime
  submittedTime
  finishedTime
  state
  submitNum
}
```

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by kinow** [comment]
Now if I accidentally misspell a variable in Cylc UI, the message should hint that there's something wrong with the variables.

![image](https://user-images.githubusercontent.com/304786/95152783-52edf400-07ea-11eb-9f7e-f1d2c6952623.png)

And fixing the variables then the query runs just fine.

![image](https://user-images.githubusercontent.com/304786/95152790-5ed9b600-07ea-11eb-868c-eb56cfc08e3b.png)
